default.mjs

```diff
@@ -241,9 +241,12 @@ export const reactConfig = [
 			"react/no-unused-prop-types": "error",
 			"react/no-unused-state": "error",
 			"react/prefer-es6-class": "error",
-			"react/require-default-props": [ "error", { ignoreFunctionalComponents: true } ],
```

Comment:
With `ignoreFunctionalComponents` being deprecated, lets use their replacement to enforce adopting the new way. Otherwise I fear we'll have a lot of default props to remove in one big sweep when it's time to switch to React 19.
```
// This will turn into an `error` in the future. It's a warning to give us time to adapt.
"react/require-default-props": [ "warn", { classes: "defaultProps", functions: "defaultArguments" } ],
```
We can set --max-warning thresholds and slowly work towards 0 warnings.

Reply:
Done with `error`. If consumers want to `warn` they can add an exemption for themselves?

Reply:
Discussed on Slack: we concluded that we slightly prefer `warn` over `error`: with `error`, all offending consumers would have to overwrite this rule, matching the arguments too. This means that we'll likely end up with 10 copies of the same rule. All copies will also need to be updated when it's time to more strictly enforce.
Having one central `warn` rule allows all consumers to just accept it and increase their max-warnings without copying and maintaining the rule definition.